### PR TITLE
tifshared imports in lambdas

### PR DESCRIPTION
**Pull request by tifapp**
Make tifshared a direct dependency of apilambda and geocodinglambda, and change backendutils imports to tifshared imports where applicable

* Import TiFShared at the root (index) of APILambda and GeocodingLambda
* Replace TiFBackendUtils imports with TiFShared imports
* Updated lambda dependencies to match TiFShared versions
* Make TiFShared a peer dependency of TiFBackendUtils to reduce duplicated dependencies https://www.johno.com/using-peer-dependencies

## Warnings
Ensure you have TiFShared linked locally or the npm install won't work. Instructions can be found in the tifshared readme https://github.com/tifapp/TiFShared?tab=readme-ov-file#local-development-setup

## Other Changes
* Added automatic TiFShared linking after local npm install
* Renamed for clarity:
  * apiCallers/events -> apiCallers/eventEndpoints
  * apiCallers/users -> apiCallers/userEndpoints
  * userFlows/events -> userFlows/createEventFlow
  * userFlows/users -> userFlows/createUserFlow
  * addPlacemarkToDB -> addLocationToDB
* Changed variable names in GeocodingLambda to align with TiFShared, ex. using the term "Location" instead of "Placemark"
* Removed duplicate test/helpers files
* Removed envVars from apilambda and instead use testEnvVars since all production envVars are in TiFBackendUtils now

https://trello.com/c/xYE5fpP3/579-tifshared-package-integration-backend

### CODE REVIEW

File: APILambda/package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
     "pr": "cd .. && npm run pr",
     "genapispecs": "ts-node --esm ./genapispecs.mjs && npx lint-openapi specs.json",
     "build": "npx tsc --strict false",
-    "test": "jest --detectOpenHandles --runInBand",
-    "test:ci": "jest --ci --detectOpenHandles --forceExit --runInBand"
+    "test": "npx ts-node node_modules/jest/bin/jest --detectOpenHandles --runInBand",
```
Comment (tifapp) [inline APILambda/package.json:10]: allows .ts files to be used in jestSetup files

**Comment by tifapp** [comment]
Linked Ticket: https://trello.com/c/xYE5fpP3

File: .github/workflows/build_api.yml [hunk 2]
```diff
@@ -53,6 +53,7 @@ jobs:
           cd APILambda
           npm ci
           npm install ../TiFBackendUtils/dist/TiFBackendUtils-1.0.0.tgz
```
Comment (seanim0920) [inline .github/workflows/build_api.yml:55]: needed to delete .ts files after building TiFShared

File: APILambda/test/helpers/events.ts
Comment (seanim0920) [inline APILambda/test/helpers/events.ts:1]: Duplicate file

**Review by suryatiru — APPROVED** [review]

**Pull request merged**